front/src/modules/timesStops/TimesStopsOutput.tsx

```diff
@@ -0,0 +1,139 @@
+import { usePathfindingV2 } from 'modules/pathfinding/hook/usePathfinding';
+import { formatSuggestedOperationalPoints } from 'modules/pathfinding/utils';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import { useFormattedOperationalPoints } from 'modules/trainschedule/hooks';
+import { getSelectedTrainId } from 'reducers/osrdsimulation/selectors';
+import { ISO8601Duration2sec, datetime2sec, secToHoursString } from 'utils/timeManipulation';
+
+import TimesStops from './TimesStops';
+import { TableType } from './types';
+import { computeMargins, findNextScheduledPoint } from './utils';
+
+type TimesStopsOutputProps = {
+  train: TrainScheduleBase;
+  simulatedTrain: SimulationResponseSuccess;
+  pathProperties: PathPropertiesFormatted;
+  infraId: number;
+};
+
+const TimesStopsOutput = ({
+  train,
+  simulatedTrain,
+  pathProperties, // raw from simulation result, calculated values
+  infraId,
+}: TimesStopsOutputProps) => {
+  const simResults = useSimulationResults();
+  usePathfindingV2(); // is it really needed to add position_on_path to pathSteps ? use position_on_path to know where the via is in the suggested point array
+  const { getPathSteps, getStartTime } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+  const startTime = useSelector(getStartTime);
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  const { data: selectedTrainSchedule = {} as TrainScheduleResult } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: selectedTrainId as number,
+      },
+      { skip: !selectedTrainId }
+    );
+
+  const trainScheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
+  const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
+    pathProperties.operationalPoints,
+    pathProperties.geometry,
+    100
+  );
+  // const sugOpPointByUic = keyBy(suggestedOperationalPoints, ({ uic, ch }) => `${uic}-${ch}`);
+  const { operationalPoints } = useFormattedOperationalPoints(
+    train,
+    simulatedTrain,
+    pathProperties,
+    infraId
+  );
+
+  const pathPointsWithOpPointIndices = selectedTrainSchedule.path.map((pathPoint) => {
+    const correspondingOpPointIndex = suggestedOperationalPoints.findIndex((sugOpPoint) => {
+      if ('uic' in pathPoint) {
+        return sugOpPoint.uic === pathPoint.uic && sugOpPoint.ch === pathPoint.secondary_code;
+      }
+      return false;
+    });
+    return {
+      ...pathPoint,
+      correspondingOpPointIndex,
+    };
+  });
+  const trainSchedulePathByUic = keyBy(pathPointsWithOpPointIndices, (pathPoint) => {
+    if ('uic' in pathPoint) {
+      return `${pathPoint.uic}-${pathPoint.secondary_code}`;
+    }
+    if ('operational_point' in pathPoint) {
+      return pathPoint.operational_point;
+    }
+    return '';
+  });
+
+  const enrichedoperationalPoints = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
```

Comment:
```suggestion
  const enrichedOperationalPoints = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
```

Reply:
Maybe you can create a new hook useOutputTableData which contains all this logic ?

Reply:
done